server/src/main/java/org/opensearch/extensions/rest/RestSendToExtensionAction.java

```diff
@@ -160,17 +164,26 @@ public String executor() {
                 return ThreadPool.Names.GENERIC;
             }
         };
+
         try {
             // Will be replaced with ExtensionTokenProcessor and PrincipalIdentifierToken classes from feature/identity
             final String extensionTokenProcessor = "placeholder_token_processor";
             final String requestIssuerIdentity = "placeholder_request_issuer_identity";
 
+            this.allowList = Set.of("Content-Type");
+            this.denyList = Set.of("Authorization", "Proxy-Authorization");
+
+            Map<String, List<String>> filteredHeaders = headers.entrySet()
```

Comment:
Can we refactor this to a method like
```java
Map<String, List<String>> filterHeaders(Map<String, List<String>>  headers, Set allowList, Set denyList) {
    // ... this line of code
}
```

If you make that utility method `public` or at least `package private` then you can test the allow/deny functionality in the test class.

Reply:
> Can we refactor this to a method like
> 
> ```java
> Map<String, List<String>> filterHeaders(Map<String, List<String>>  headers, Set allowList, Set denyList) {
>     // ... this line of code
> }
> ```
> 
> If you make that utility method `public` or at least `package private` then you can test the allow/deny functionality in the test class.

Should it be added to RestSendToExtensionAction or to another class?